OneDrivePhotoBrowser/AccountSelection.xaml.cs

```diff
@@ -51,28 +64,33 @@ public AccountSelection()
 
         private async void AccountSelection_Loaded(object sender, RoutedEventArgs e)
         {
-            if (((App)Application.Current).OneDriveClient != null)
+            var app = ((App) Application.Current);
+            if (app.OneDriveClient != null)
             {
-                await ((App)Application.Current).OneDriveClient.SignOutAsync();
-
-                var client = ((App)Application.Current).OneDriveClient as OneDriveClient;
-                if (client != null)
+                var msaAuthProvider = app.AuthProvider as MsaAuthenticationProvider;
+                var adalAuthProvider = app.AuthProvider as AdalAuthenticationProvider;
```

Comment:
>AuthProvider [](start = 43, length = 12)

Isn't there a shared interface between these AuthProviders so this can be rewritten as await app.AuthProvider.SignOutAsync()?